File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/data/config/impl/PlayerConfig.java
package com.dkellycollins.triggerfinger.data.config.impl;

import android.content.Context;

import com.dkellycollins.triggerfinger.R;
import com.dkellycollins.triggerfinger.data.config.IPlayerConfig;

public class PlayerConfig extends RConfig implements IPlayerConfig {

    public PlayerConfig(Context context) {
        super(context);
    }

    @Override
    public int getBitmapId() {
        return R.drawable.player_bitmap;
    }

    @Override
    public float getCollidableRadius() {
        return getFloat(R.integer.player_collidable_radius);
    }
}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/data/config/impl/PlayerConfig.kt
package com.dkellycollins.triggerfinger.data.config.impl

import android.content.Context

import com.dkellycollins.triggerfinger.R
import com.dkellycollins.triggerfinger.data.config.IPlayerConfig

class PlayerConfig(context: Context) : RConfig(context), IPlayerConfig {

    override val bitmapId: Int
        get() = R.drawable.player_bitmap

    override val collidableRadius: Float
        get() = getFloat(R.integer.player_collidable_radius)

    override val invincibleTime: Int
        get() = _context.resources.getInteger(R.integer.player_invincible_time)
}
